src/black/__init__.py

```diff
@@ -515,8 +520,21 @@ def get_sources(
     stdin_filename: Optional[str],
 ) -> Set[Path]:
     """Compute the set of files to be formatted."""
+    root, method = find_project_root(src)
+
+    if verbose:
+        if method:
+            out(
+                f"Identified `{root}` as project root containing a {method}.",
+                fg="blue",
+            )
+        else:
+            out(f"Identified `{root}` as project root.", fg="blue")
+        paths = '", "'.join(
+            str(Path(source).resolve().relative_to(root)) for source in src
+        )
```

Comment:
Would it be worth it to guard this with a try... except? When we normalize path before applying the file discovery regexes (using `normalize_path_maybe_ignore`) we have be able to handle permission or out of tree exceptions gracefully and this code does none of that. Not sure how this would look but I'd prefer a defensive approach to our filepath handing code.